src/realm/object-store/audit.hpp

```diff
@@ -16,20 +16,109 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
+#ifndef REALM_AUDIT_HPP
+#define REALM_AUDIT_HPP
+
+#include <realm/util/functional.hpp>
+#include <realm/util/optional.hpp>
+#include <realm/util/terminate.hpp>
+
+#include <memory>
+#include <string>
+#include <vector>
+
 namespace realm {
-class Table;
+class DB;
+class AuditObjectSerializer;
+class Obj;
+class SyncUser;
 class TableView;
-template <typename>
-class BasicRowExpr;
-using RowExpr = BasicRowExpr<Table>;
+class Timestamp;
+struct ColKey;
+struct RealmConfig;
+struct SyncError;
 struct VersionID;
+namespace util {
+class Logger;
+}
+
+struct AuditConfig {
+    // User to open the audit Realms with. If null, the user used to open the
+    // Realm being audited is used
+    std::shared_ptr<SyncUser> audit_user;
+    // Prefix added to the start of generated partition keys for audit Realms.
+    std::string partition_value_prefix = "audit";
+    // Object serializer instance for converting objects to JSON payloads. If
+    // null, a default implementation is used.
+    std::shared_ptr<AuditObjectSerializer> serializer;
+    // Logger for audit events. If null, the sync logger from the Realm under
+    // audit is used.
+    std::shared_ptr<util::Logger> logger;
+    // Error handler which is called if fatal sync errors occur on the sync
+    // Realm. If null, an error is logged then abort() is called.
+    std::function<void(SyncError)> sync_error_handler;
+    // Metadata to attach to each audit event. Each key used must be a property
+    // in the server-side schema for AuditEvent. This is not validated and will
+    // result in a sync error if violated.
+    std::vector<std::pair<std::string, std::string>> metadata;
+};
 
 class AuditInterface {
 public:
-    virtual ~AuditInterface() {}
+    virtual ~AuditInterface() = default;
```

Comment:
This originally was a pure virtual interface in the open source repo with the implementation living in the private repo. That division is no longer needed, but I ended up leaving it because otherwise this header has to pull in the entire world.